bin/SConsDoc.py

```diff
@@ -859,45 +851,23 @@ def parseXmlFile(self, fpath):
         # Parse it
         self.parseDomtree(t.root, t.xpath_context, t.nsmap)
 
-# lifted from Ka-Ping Yee's way cool pydoc module.
-if PY2:
-    def importfile(path):
-        """Import a Python source file or compiled file given its path."""
-        import imp
-        magic = imp.get_magic()
-        with open(path, 'r') as ifp:
-            if ifp.read(len(magic)) == magic:
-                kind = imp.PY_COMPILED
-            else:
-                kind = imp.PY_SOURCE
-        filename = os.path.basename(path)
-        name, ext = os.path.splitext(filename)
-        with open(path, 'r') as ifp:
-            try:
-                module = imp.load_module(name, ifp, path, (ext, 'r', kind))
-            except ImportError as e:
-                sys.stderr.write("Could not import %s: %s\n" % (path, e))
-                return None
-        return module
-
-else:  # PY3 version, from newer pydoc
-    def importfile(path):
-        """Import a Python source file or compiled file given its path."""
-        from importlib.util import MAGIC_NUMBER
-        with open(path, 'rb') as ifp:
-            is_bytecode = MAGIC_NUMBER == ifp.read(len(MAGIC_NUMBER))
-        filename = os.path.basename(path)
-        name, ext = os.path.splitext(filename)
-        if is_bytecode:
-            loader = importlib._bootstrap_external.SourcelessFileLoader(name, path)
-        else:
-            loader = importlib._bootstrap_external.SourceFileLoader(name, path)
-        # XXX We probably don't need to pass in the loader here.
-        spec = importlib.util.spec_from_file_location(name, path, loader=loader)
-        try:
-            return importlib._bootstrap._load(spec)
-        except ImportError:
-            raise ErrorDuringImport(path, sys.exc_info())
+def importfile(path):
```

Comment:
looks like we need an import of `importlib` (preferably at the top), don't know why this hasn't been a problem before - several sider complaints.